src/internal/observable/makeConnectable.ts

```diff
@@ -0,0 +1,43 @@
+/** @prettier */
+
+import { ObservableInput } from '../types';
+import { Subject } from '../Subject';
+import { Subscription } from '../Subscription';
+import { Observable } from '../Observable';
+import { defer } from './defer';
+
+export type ConnectableObservableLike<T> = Observable<T> & {
+  /**
+   * (Idempotent) Calling this method will connect the underlying source observable to all subscribed consumers
+   * through an underlying {@link Subject}.
+   * @returns A subscription, that when unsubscribed, will "disconnect" the source from the connector subject,
+   * severing notifications to all consumers.
+   */
+  connect(): Subscription;
+};
+
+/**
+ * Creates an observable that multicasts once `connect()` is called on it.
+ *
+ * @param source The observable source to make connectable.
+ * @param connector The subject to used to multicast the source observable to all subscribers.
+ * Defaults to a new {@link Subject}.
+ * @returns A "connectable" observable, that has a `connect()` method, that you must call to
+ * connect the source to all consumers through the subject provided as the connector.
+ */
+export function makeConnectable<T>(source: ObservableInput<T>, connector: Subject<T> = new Subject<T>()): ConnectableObservableLike<T> {
+  let connection: Subscription | null = null;
+
+  const result: any = new Observable<T>((subscriber) => {
+    return connector.subscribe(subscriber);
+  });
+
+  result.connect = function () {
+    if (!connection) {
+      connection = defer(() => source).subscribe(connector);
```

Comment:
question: Is there a reason for using `defer` here instead of `from`? `source` is a parameter and is never reassigned, so I don't see why this needs to be lazy.